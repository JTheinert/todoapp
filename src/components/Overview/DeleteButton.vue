<template>
  <button
    :id="index"
    class="delete-button"
    @click="openModal(todo)">
    <img
      src="../assets/images2.png"
      class="image" >
  </button>
</template>

<script>
export default{
  props: {
    index: {
      type: Number,
      default: 0

    }
  },
  data () {
    return {
      todo: this.$store.state.todos[this.$route.params.id]
    }
  },
  methods: {
    openModal (todo) {
      // eslint-disable-next-line no-unused-expressions
      this.$router.push('delete')
      this.$store.commit('updateNummer', {
        nummer: todo.id
      })
    }
  }
}
</script>
<style>
.delete-button {
background-color: red;
font-size: 20px;
border-radius: 10px;
text-align: right;
color: white;
}
.image{
margin: 5px;
height: 40px;
}
</style>
