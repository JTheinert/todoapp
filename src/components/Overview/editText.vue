<template>
  <p class="text">
    {{ text }}
  </p>
</template>
<script>
export default {
  props: {
    text: {
      type: String,
      default: 'You can edit your todo with a just one click!'
    }

  }
}
</script>
<style >
.text{
margin-right: 10px;
display: flex;
font-size: 15px;
text-align: center;
justify-content: center;
align-items: center;
}
</style>
