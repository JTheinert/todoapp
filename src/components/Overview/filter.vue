<template>
  <div class="filter">
    <a
      v-if="filteredNr == 1"
      name="All1"
      style="text-decoration: underline;"
      @click="filter()"
    >All</a
    >
    <a
      v-else
      name="All2"
      @click="filter()">All</a>
    <a
      v-if="filteredNr == 2"
      name="notDone1"
      style="text-decoration: underline;"
      @click="filterNotDone()"
    >in Progress</a
    >
    <a
      v-else
      @click="filterNotDone()">in Progress</a>

    <a
      v-if="filteredNr == 3"
      style="text-decoration: underline;"
      @click="filterDone()"
    >Done</a
    >
    <a
      v-else
      @click="filterDone()">Done</a>
  </div>
</template>

<script>
export default{
  computed: {

    filteredNr () {
      return this.$store.state.filteredNr
    }
  },
  methods: {
    filter () {
      this.$store.state.filteredNr = 1
    },

    filterNotDone () {
      this.$store.state.filteredNr = 2
    },

    filterDone () {
      this.$store.state.filteredNr = 3
    }
  }
}
</script>

<style>
.filter {
display: flex;
margin-left: 20px;
margin-right: 20px;
color: #2a9cac;
justify-content: space-around;
margin-bottom: 10px;
}
</style>
