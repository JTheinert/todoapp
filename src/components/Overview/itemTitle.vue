<template>
  <p class="item-title">{{ todo.title }}</p>
</template>

<script>
export default{
  data () {
    return {
      todo: this.$store.state.todos[this.$route.params.id]
    }
  }
}
</script>

<style>
.item-title{
color:green;
justify-content: left;
display: flex;
font-size: 35px;
}
</style>
