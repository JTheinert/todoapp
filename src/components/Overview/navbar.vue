<template>
  <div>
    <a
      style="text-decoration: underline; color: #2447b9; margin-right: 20px; margin-left: 20px; border: 5px;"
      @click="home()"
    >Home</a
    >
    <a
      style="text-decoration: underline; color: #2447b9;margin-right: 20px; margin-left: 20px; border: 5px;"
      @click="deletePage()"
    >Delete</a
    >
    <a
      style="text-decoration: underline; color: #2447b9;margin-right: 20px; margin-left: 20px; border: 5px;"
      @click="testing()"
    >testing</a
    >
  </div>
</template>

<script>
export default{
  methods: {
    home () {
      this.$router.push('home')
    },
    deletePage () {
      this.$router.push('delete')
    },
    testing () {
      this.$router.push('testing')
    }
  }
}
</script>

<style>

</style>
