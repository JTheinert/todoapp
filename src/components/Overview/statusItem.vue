<template>
  <div>
    <div>
      <div
        v-if="todo.used == 1"
        class="status-item">
        <p style="color:darkgoldenrod;">in progress</p>
      </div>
      <div
        v-if="todo.used == 2"
        class="status-item">
        <p class="done">Done</p>
      </div>
    </div>
  </div>

</template>
<script>
export default {
  data () {
    return {
      todo: this.$store.state.todos
    }
  },
  computed: {

  }
}
</script>
<style >
.status-item {
display: block;
align-items: left;
border: 1px;
font-size: 20px;
text-align: center;
}
.done {
color: green;
}
.progress {
color: darkgoldenrod;
}
</style>
