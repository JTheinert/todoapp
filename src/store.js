/* eslint-disable no-console */
import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export const store = new Vuex.Store({
  mutations: {
    updateIsOpen (state, { id, isOpen }) {
      const todo = state.todos.find(obj => obj.id === id)
      todo.isOpen = isOpen
    },
    updateFilteredNr () {
      // eslint-disable-next-line no-use-before-define
      const filteredNr = filteredNr
    },

    set (newTodo) {
      return newTodo
    },
    updateNummer (state, { nummer }) {
      state.nummer = nummer
    },
    updateIdForTodo (state) {
      state.idForTodo++
    }
  },
  methods: {
    mounted () {
      // eslint-disable-next-line no-console
      console.log('App mounted!')
      if (localStorage.getItem('todos')) { this.todos = JSON.parse(localStorage.getItem('todos')) }
    },
    watch: {
      todos: {
        handler () {
          console.log('Todos changed!')
          localStorage.setItem('todos', JSON.stringify(this.todos))
        },
        deep: true
      }
    }
  },
  state: {
    filteredNr: 1,
    modalShow: false,
    newTodo: '',
    idForTodo: 1,
    beforeEditCache: '',
    beforeEditDescriptionCache: '',
    placeholder: 'Edit Description',
    dialog: false,
    nummer: 0,
    todos: [
      {
        id: 0,
        title: 'cool',
        editing: false,
        descriptionEditing: false,
        isOpen: false,
        status: "isn't done",
        description: '',
        filter: false,
        filterNr: 1,
        used: 3
      }
    ]
  },

  getters: {
    doneTodos: state => {
      return state.todos.filter(todo => todo.done)
    },
    doneTodosCount: (getters) => {
      return getters.doneTodos.length
    },
    getTodoById: state => id => {
      return state.todos.find(todo => todo.id === id)
    },
    getName: () => 'rajesh'
  }
})
