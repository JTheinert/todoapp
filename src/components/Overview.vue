<template>
  <div>
    <!-- THIS IS THE NEW TODO INPUT -->
    <navbar/>

    <input
      v-focus
      id="1001"
      v-model="newTodo"
      type="text"
      class="todo-input"
      placeholder="Todo Here!"
      @keyup.enter="addTodo"
    >
    <filterung/>

    <todo-item-middle/>

  </div>
</template>

<script>
import navbar from './Overview/navbar'
import filterung from './Overview/filter'
import todoItemMiddle from './Overview/todoItemMiddle'

export default {
  name: 'Overview',
  components: { navbar, filterung, todoItemMiddle },

  directives: {
    focus: {
      inserted: function (el) {
        el.focus()
      }
    }
  },

  data () {
    return {
      newTodo: ''
    }
  },
  computed: {
    todoList () {
      return this.$store.state.todos
    },
    idForTodo () {
      return this.$store.state.idForTodo
    }
  },

  methods: {
    addTodo () {
      if (this.newTodo.length === 0) {
        return
      }

      this.todoList.push({
        id: this.idForTodo,
        title: this.newTodo,
        editing: false,
        descriptionEditing: false,
        isOpen: false,
        buttonNme: 'Detils',
        sttus: "isn't done",
        description: '',
        filter: false,
        filterNr: 1
      });
      // eslint-disable-next-line no-unused-expressions
      (this.newTodo = '')
      // eslint-disable-next-line no-unused-expressions
      this.$store.commit('updateIdForTodo')
      localStorage.setItem('0', 'isnt done')
    },

    editTodo (todo) {
      todo.editing = true
      this.beforeEditCche = todo.title
    },
    doneEdit (todo) {
      if (todo.title.trim().length === 0) {
        todo.title = this.beforeEditCche
      }
      todo.editing = false
      this.$store.commit('updateTitle', {
        id: todo.id,
        title: todo.title
      })
    },
    cancelEdit (todo) {
      todo.title = this.beforeEditCche
      todo.editing = false
    },

    openModal (todo) {
      // eslint-disable-next-line no-unused-expressions
      this.$router.push('delete')
      this.$store.commit('updateNummer', {
        nummer: todo.id
      })
    }

  }
}
</script>

<style>
.todo-input {
width: 100%;
padding: 10px 250px;
font-size: 20px;
margin-bottom: 16px;
color: #29cc;
text-align: center;
margin: 10px;
border: 2px solid lavenderblush;
}
.todo-item {
margin-bottom: 3px;
display: inline-block;
align-items: center;
justify-content: space-between;
animation-duration: 0.3s;
color: #29cc;
border: 1px solid lavenderblush;
width: 100%;
}

.todo-item-left {
display: flex;
align-items: left;
}

</style>
