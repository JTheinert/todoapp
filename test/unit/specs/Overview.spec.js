import { mount, createLocalVue } from '@vue/test-utils'
import Overview from '@/src/Components/Overview.vue'

import Vuex from 'vuex'
import { store } from '@/store'
import routerLink from 'vue-router'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Overview.vue', () => {
  test('Test', () => {
    expect(true).toBe(true)
  })
  test('function has been called', () => {
    const wrapper = mount(Overview, { localVue, store, routerLink })

    const openModal = jest.fn()

    wrapper.setMethods({
      openModal: openModal
    })
    wrapper.vm.$nextTick()
    wrapper.find('button.delete-button').trigger('click')

    expect(openModal).toHaveBeenCalled()
  })

  // const wrapper = mount(Overview)
  // it('has a button', () => {
  //     expect(wrapper.contains('header')).toBe(true)
  //   })

  test('Title ?', () => {
    const wrapper = mount(Overview, { localVue, store, routerLink })
    expect(wrapper.text()).toContain('cool')
  })

  test('All ?', () => {
    const wrapper = mount(Overview, { localVue, store, routerLink })
    expect(wrapper.text()).toContain('All')
  })

  test('in Progress ?', () => {
    const wrapper = mount(Overview, { localVue, store, routerLink })
    expect(wrapper.text()).toContain('in Progress')
  })

  test('Done ?', () => {
    const wrapper = mount(Overview, { localVue, store })
    expect(wrapper.text()).toContain('Done')
  })
  test('renders correctly', () => {
    const wrapper = mount(Overview, { store, localVue, routerLink })
    expect(wrapper.element).toMatchSnapshot()
  })
})
