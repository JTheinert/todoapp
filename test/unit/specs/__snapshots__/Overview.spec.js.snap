// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview.vue renders correctly 1`] = `
<div>
  <a
    style="text-decoration: underline; color: rgb(36, 71, 185); margin-right: 20px; margin-left: 20px; border: 5px;"
  >
    Home
  </a>
   
  <a
    style="text-decoration: underline; color: rgb(36, 71, 185); margin-right: 20px; margin-left: 20px; border: 5px;"
  >
    Delete
  </a>
   
  <a
    style="text-decoration: underline; color: rgb(36, 71, 185); margin-right: 20px; margin-left: 20px; border: 5px;"
  >
    testing
  </a>
   
  <input
    class="todo-input"
    id="1001"
    placeholder="Todo Here!"
    type="text"
  />
   
  <div
    class="filter"
  >
    <a
      name="All1"
      style="text-decoration: underline;"
    >
      All
    </a>
     
    <a>
      in Progress
    </a>
     
    <a>
      Done
    </a>
  </div>
   
  <div
    class="todo-item"
    id="0"
  >
    <div
      class="todo-item-middle"
    >
      <div
        class="todo-item-label"
      >
        <div
          class="todo-item-title"
        >
          
          cool
        
        </div>
         
        <!---->
         
        <!---->
         
        <p
          class="teaser-text"
        >
          
          
        
        </p>
      </div>
    </div>
     
    <div
      class="todo-item-right"
    >
      <p
        style="margin-right: 10px; display: flex; font-size: 15px; text-align: center; justify-content: center; align-items: center;"
      >
        
        You can edit your todo with a just one click!
      
      </p>
       
      <button
        class="delete-button"
        id="0"
      >
        <img
          class="image"
          src="./assets/images2.png"
        />
      </button>
    </div>
  </div>
</div>
`;
